dump/dump_test.go

```diff
@@ -0,0 +1,456 @@
+/*
+Copyright 2017 Google Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package dump
+
+import (
+	"testing"
+)
+
+func TestSdumpPrimitives(t *testing.T) {
+	testcases := []struct {
+		name     string
+		input    func() interface{}
+		expected string
+	}{
+		{
+			"boolTrue",
+			func() interface{} {
+				return true
+			},
+			"var Obj = true\n",
+		},
+		{
+			"boolFalse",
+			func() interface{} {
+				return false
+			},
+			"var Obj = false\n",
+		},
+		{
+			"int",
+			func() interface{} {
+				return 10
+			},
+			"var Obj = 10\n",
+		},
+		{
+			"int8",
+			func() interface{} {
+				return int8(10)
+			},
+			"var Obj = 10\n",
```

Comment:
The type is changing here.  Is that OK?  You could preserve the type, but if you don't need to, you could just document this instead.